client-js/app/elements/labrad-registry.ts

```diff
@@ -61,31 +53,49 @@ export class LabradRegistry extends polymer.Base {
     return item.name.match(this.regex);
   }
 
-  selectKey() {
-    this.selDir = null;
-    this.selectType = 'key';
+  @listen('idx-changed')
+  changedSelected() {
+    console.log('ChangeSelected', this.idx);
   }
 
-  selectDir() {
-    this.selKey = null;
-    this.selectType = 'dir';
+  @computed()
+  selected(idx: number): boolean {
+    console.log("computing selected is: ", idx);
+    if (idx != null) {
+        return false;
+      } else {
+        console.log('doesnt exist');
+        return true;
+      }
+  }
+  @property({computed: '_computeSelectedType(idx)'})
+  selectedType: string;
```

Comment:
Yes, if I put it in the ```if``` statements on ```doCopy()``` for example. Also, they're passed to the RPC calls on line 375, for example. Those generate compiler error.